crates/protect-ffi/src/encrypt_config.rs

```diff
@@ -0,0 +1,490 @@
+use super::Error;
+use cipherstash_client::schema::{
+    column::{Index, IndexType, TokenFilter, Tokenizer},
+    ColumnConfig, ColumnType,
+};
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, str::FromStr};
+
+#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+pub struct Identifier {
+    #[serde(rename = "t")]
+    pub table: String,
+    #[serde(rename = "c")]
+    pub column: String,
+}
+
+impl Identifier {
+    pub fn new<S>(table: S, column: S) -> Self
+    where
+        S: Into<String>,
+    {
```

Comment:
Be aware that this trait bound means both arguments must be the same type. If you want to allow any type for either (which I think is the intent), you should do:

```rust
pub fn new(table: impl Into<String>, column: impl Into<String>) -> Self
```

Or the more verbose (and also requires generic parameters if they can't be inferred):

```rust
pub fn new<T, C>(table: T, column: C) -> Self
    where
        T: Into<String>,
        C: Into<String>,
```

Reply:
Ah, fair. This was brought over from Proxy, but protect-ffi doesn't really need the generics. I'll refactor to remove the generics.

Reply:
Eh, turns out that the generics are convenient in tests. Might leave this one as is and update if it become an issue that the args can't be the same type.